src/Inputs/SearchBar.tsx

```diff
@@ -0,0 +1,268 @@
+                            onClick={onSelect}
+                            key={val as string}
+                            order="1"
+                        />
+                    );
+                }
+            } else if (
+                child &&
+                isValidElement(child) &&
+                inputValue &&
+                child.props.children
+                    .toLowerCase()
+                    .search(inputValue.toLowerCase()) > -1
+            ) {
+                const val = child.props.value;
+                const selected = String(value) === val;
+                const order = child.props.children
+                    .toLowerCase()
+                    .startsWith(inputValue.toLowerCase())
+                    ? REVERSE
+                    : '1';
+                return (
+                    <SelectItem
+                        {...child.props}
+                        active={selected}
+                        onClick={onSelect}
+                        key={val as string}
+                        order={order as string}
+                    />
+                );
+            }
+
+            return null;
+        });
+
+    const onSelect = useCallback(
+        ({ currentTarget }): void => {
+            onChange({
+                target: {
+                    value: currentTarget.getAttribute('value'),
+                    name,
+                },
+            });
+            setinputValue(currentTarget.firstChild.nodeValue);
+        },
+        [name],
+    );
+
+    const options = useMemo((): any => {
+        return Children.toArray(children);
+    }, [expanded]);
+
+    useLayoutEffect((): void | (() => void) => {
+        if (refSelectList.current?.children.length) {
+            setNumVisibleSelection(refSelectList.current.children.length);
+        } else {
+            setNumVisibleSelection(0);
+        }
+
+        const listener = (): void => {
+            setExpanded(false);
+        };
+
+        window.addEventListener('keydown', listener, { once: true });
+        window.addEventListener('click', listener, { once: true });
+
+        return (): void => {
+            window.removeEventListener('keydown', listener);
+            window.removeEventListener('click', listener);
+        };
+    }, [expanded]);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setinputValue(event.currentTarget.value);
+        onInput(event.currentTarget.value);
+        setExpanded(true);
+    };
+    return (
+        <LabelLayout {...props}>
+            <Container>
+                <SelectDisplay>
+                    <InputFragment
+                        {...props}
+                        value={inputValue}
+                        placeholder={placeholder}
+                        onChange={(e): void => handleChange(e)}
+                    />
+                    <Icon />
+                </SelectDisplay>
+                {mount && (
```

Comment:
There's alot of logic for this, consider putting it in another component

Reply:
@fpunny sorry I didn't understand this one

Reply:
@sahandfarahani He is just saying to seperate all the logic tied to everything inside the {} brackets to a new component to make it more readable